controlplane/pkg/model/model.go

```diff
@@ -35,7 +36,7 @@ type Model interface {
 	GetEndpoint(name string) *registry.NSERegistration
 	AddEndpoint(endpoint *registry.NSERegistration)
 	DeleteEndpoint(name string) error
-	SelectEndpoint(nsName string) (*registry.NSERegistration, error)
+	SelectEndpoint(nsName string, nsLabels *map[string]string) (*registry.NSERegistration, error)
```

Comment:
So I'd actually suggest

SelectEndpoing(*registry.NetworkService)

that way you get the name, the matches, the destinations etc :)

Reply:
I'm not sure how this can be invoked in this case. Please check my comment in `network_service_server.go` below.